zerver/views/realm.py

```diff
@@ -61,7 +61,7 @@ def update_realm(
         allow_edit_history: Optional[bool]=REQ(validator=check_bool, default=None),
         default_language: Optional[str]=REQ(validator=check_string, default=None),
         waiting_period_threshold: Optional[int]=REQ(converter=to_non_negative_int, default=None),
-        authentication_methods: Optional[Dict[Any, Any]]=REQ(validator=check_dict([]), default=None),
+        authentication_methods: Optional[Dict[str, Any]]=REQ(validator=check_dict([]), default=None),
```

Comment:
We can probably type this more fully, if we're touching this line; it should be a `Dict[str, bool]`.

Reply:
I want to leave the value type here as  `Any` until we have confidence that the validator actually makes sure that we get a `bool` here, which it clearly doesn't now.  I think all the `check_dict` validators are kinda suspect and out of the scope of my PR, and I think @andersk has a plan for them.

Reply:
ok